pkg/controller/constrainttemplate/constrainttemplate_controller.go

```diff
@@ -441,8 +441,14 @@ func (r *ReconcileConstraintTemplate) handleUpdate(
 	t.Observe(unversionedCT)
 
 	generateVap, err := constraint.ShouldGenerateVAP(unversionedCT)
-	if err != nil && !errors.Is(err, celSchema.ErrCodeNotDefined) {
+	if err != nil {
 		logger.Error(err, "generateVap error")
+		if generateVap {
```

Comment:
by Golang convention, `generateVap` would be false if ShouldGenerateVAP returns an error. We should probably report the error regardless.

Reply:
This would mean all rego templates will have this status message regarldess of if there is intention of VAP generation or not. When should we report this error?
- all the time, regardless of indication of VAP generation for CT?
- Or only when VAP is supposed to be generated but is not being generated?

I agree that we will be reporting this error for all rego CT when we go to beta.

Reply:
It sounds like `ShouldGenerateVAP` should be implemented differently then?

If we are struggling to implement the behaviors we want, that's a sign we should rethink primitives, not double-down on pre-existing shapes as if they were carved in stone, adding edge cases and complexity along the way.

It seems like `ShouldGenerateVAP()` should return (`false`, `nil`) if we get `ErrCodeNotDefined`:

https://github.com/open-policy-agent/gatekeeper/blob/3555a65da7665e3e510a0941c5093cfd80f9d84f/pkg/drivers/k8scel/schema/schema.go#L291

Reply:
Updated `shouldGenerateVAP` to return false. Still returning `ErrCodeNotDefined` as it is needed for constraints to report respective status.

@ritazh @sozercan PTAL.